src/name.rs

```diff
@@ -399,12 +399,61 @@ pub(crate) struct NamespaceResolver {
     nesting_level: i32,
 }
 
+/// These constants define the [reserved namespaces] for the xml standard.
+///
+/// The prefix `xml` is by definition bound to the namespace name
+/// `http://www.w3.org/XML/1998/namespace`. It may, but need not, be declared, and must not be
+/// undeclared or bound to any other namespace name. Other prefixes must not be bound to this
+/// namespace name, and it must not be declared as the default namespace.
+///
+/// The prefix `xmlns` is used only to declare namespace bindings and is by definition bound
+/// to the namespace name `http://www.w3.org/2000/xmlns/`. It must not be declared or
+/// undeclared. Other prefixes must not be bound to this namespace name, and it must not be
+///  declared as the default namespace. Element names must not have the prefix `xmlns`.
+///
+/// [reserved namespaces]: https://www.w3.org/TR/xml-names11/#xmlReserved
+const RESERVED_NAMESPACE_XML: (Prefix, Namespace) = (
+    Prefix(b"xml"),
+    Namespace(b"http://www.w3.org/XML/1998/namespace"),
+);
+const RESERVED_NAMESPACE_XMLNS: (Prefix, Namespace) = (
+    Prefix(b"xmlns"),
+    Namespace(b"http://www.w3.org/2000/xmlns/"),
+);
+
+const RESERVED_NAMESPACES: [(Prefix, Namespace); 2] =
+    [RESERVED_NAMESPACE_XML, RESERVED_NAMESPACE_XMLNS];
+
 impl NamespaceResolver {
+    pub fn new_root_resolver_and_buf() -> (Self, Vec<u8>) {
```

Comment:
I have *not* done a thorough search and @Mingun is the expert on this code not me, but it seems that the buffer is always? used alongside the `NamespaceResolver`, in which case, is there a reason it can't be part of the `NamespaceResolver` rather than being passed around?

(I may have missed somewhere where it was being used elsewhere)

Reply:
I was wondering the same, but I really didn't have an interest in making a change to that design as part of this change.

Reply:
> I have _not_ done a thorough search and @Mingun is the expert on this code not me, but it seems that the buffer is always? used alongside the `NamespaceResolver`, in which case, is there a reason it can't be part of the `NamespaceResolver` rather than being passed around?

Yes, I also noticed this and will change this in that PR